File: day01/day01.js
function inverse_captcha(data, shift) {
    return data.filter((n, i, a) => n === a[(i + shift) % a.length])
               .reduce((sum, n) => sum + n, 0);
}

const data = [...document.getElementById("input").textContent.trim()].map(Number);
document.writeln();
document.writeln(inverse_captcha(data, 1));
document.writeln(inverse_captcha(data, data.length / 2));

File: day01/day01.clj
(require '[clojure.string :as str])
(defn inverse-captcha [data shift]
  (->> data
       (map vector (vec (concat (subvec data shift) (subvec data 0 shift))))
       (filter #(= (nth % 0) (nth % 1)))
       (map #(nth % 0))
       (reduce +)))

(def data (vec (map #(Character/digit % 10) (str/trim (slurp "input.txt")))))
(println (inverse-captcha data 1))
(println (inverse-captcha data (/ (count data) 2)))
